### PR TITLE
docs: :building_construction: replace with `create_properties_object()`

### PR DESCRIPTION
**Pull request by signekb**
## Description

Replacing `create_package_properties()` used internally in `create_package_structure()`.

## Reviewer Focus

This PR needs a quick review.

## Checklist

- [X] Updated documentation

### CODE REVIEW

**Review by martonvago — COMMENTED** [review]
Makes sense, just some questions/points that we don't necessarily have to answer for this PR:
- What is `create_properties_object` doing with the `path` argument?
- We'll have to think about whether we want to pass our properties around as dataclasses or as dictionaries (= json objects). My feeling is to go for dataclasses because then we benefit from their reliable structure and can put any package or resource-specific logic in those classes. But there may be arguments against this?
- I think it makes sense to include this step in the diagram as a conceptual step, but in terms of the actual code, this might just turn out to be a class instantiation (i.e. `package = Package(id=id)` or `package = asdict(Package(id=id))` if we want a dictionary)?

**Review by lwjohnst86 — CHANGES_REQUESTED** [review]
@martonvago good observations!

For the second point, I agree, I think data classes would be right way to go and I don't see any strong arguments against it.

For your third point, I think that isn't necessary given the text of `create_properties_object()`, but it could be made a bit clearer to say "properties dataclass is initialized internally" or something.

File: docs/design/implementation/images/core/create-package-structure.puml [hunk 1]
```diff
@@ -44,10 +44,10 @@ state "create_package_structure()" as cps {
   cd : - Output: A list of paths\n  of created directories
   cip --> cd
 
-  state "create_package_properties()" as cppt
+  state "create_properties_object()" as cppt
   cppt : - Required arg: path
   cppt : - Output: JSON object
-  cppt : - Uses create_properties_template()\n  internally
+  cppt : - Uses properties dataclasses\n  internally and sets default values
   cd --> cppt
```
[the remark this answers is not in the document]
Comment (signekb) [reply docs/design/implementation/images/core/create-package-structure.puml:51]: @lwjohnst86 I'm not sure what you mean here. Should I remove `create_properties_object()` from this diagram completely? And then we'll just initialise the dataclass with default values within `create_package_structure()`? Or do we want a function for that initialisation?

File: docs/design/implementation/images/core/create-package-structure.puml [hunk 2]
```diff
@@ -44,10 +44,10 @@ state "create_package_structure()" as cps {
   cd : - Output: A list of paths\n  of created directories
   cip --> cd
 
-  state "create_package_properties()" as cppt
+  state "create_properties_object()" as cppt
   cppt : - Required arg: path
```
Comment (lwjohnst86) [inline docs/design/implementation/images/core/create-package-structure.puml:48]:
```suggestion
```

File: docs/design/implementation/images/core/create-package-structure.puml [hunk 3]
```diff
@@ -44,10 +44,10 @@ state "create_package_structure()" as cps {
...
   cppt : - Output: JSON object
-  cppt : - Uses create_properties_template()\n  internally
+  cppt : - Uses properties dataclasses\n  internally and sets default values
   cd --> cppt
```
Comment (lwjohnst86) [inline docs/design/implementation/images/core/create-package-structure.puml:52]:
```suggestion

```

**Review by lwjohnst86 — CHANGES_REQUESTED** [review]
@signekb these suggestions hopefully make it clearer what I meant ☺️

**Comment by signekb** [comment]
@lwjohnst86 Amazing, yes, that definitely made it clearer. Thanks! 
What about the outputs and inputs of JSON objects in this diagram? Should that be changed to the `Properties` class?

**Comment by lwjohnst86** [comment]
@signekb for now, we'll refer to outputs/inputs as JSON, but this might change once we finalize and implement the properties as dataclasses

**Review by lwjohnst86 — APPROVED** [review]

**Pull request merged**